.gitignore

```diff
@@ -2,3 +2,4 @@
 dist-newstyle/
 cabal.project.local*
 *.buildinfo
+.hie
```

Comment:
maybe we want to add `*.hiedb` too

Reply:
I switched to putting all test output (ghc output, like .hi, .o and .hie files) into `test/tmp` which is removed after tests, so shouldn't need to add anything to .gitignore